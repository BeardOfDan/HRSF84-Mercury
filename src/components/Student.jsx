import React from 'react';

const Student = ({studentData}) => {
  return (
    <div className="student list-item container column">
      <a className="list-item-a student-a" href={`/profile/${studentData._id}`}>
        <img className="list-item-img student-img" src={studentData.picture} />
        <label className="list-item-name student-name">{studentData.name}</label>
      </a>
    </div>
  )
};

export default Student;
