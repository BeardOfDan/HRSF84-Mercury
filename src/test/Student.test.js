import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Student from '../components/Student';

Enzyme.configure({ adapter: new Adapter() });

describe('Student', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Student />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});