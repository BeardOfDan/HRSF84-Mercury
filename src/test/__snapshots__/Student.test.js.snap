// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student should render correctly with props 1`] = `
<div
  className="student container column"
>
  <a
    className="student-box"
    href="/profile/5a0d089080bcabc5ac8cb222"
  >
    <img
      className="list-item-img student-img"
      src="https://www.w3schools.com/w3css/img_fjords.jpg"
    />
    <label
      className="list-item-name student-name"
    >
      Frodo Baggins
    </label>
  </a>
</div>
`;
